### PR TITLE
Support windows logging

### PR DESCRIPTION
**Pull request by aiwantaozi**
Problem:
windows cluster not support logging yet

Solution: 
add build windows fluentd image related files

### CODE REVIEW

File: package/windows/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+RUN choco install -y ruby --version 2.4.2.2 --params "'/InstallDir:C:\'" \
+&& choco install -y msys2 --params "'/NoPath /NoUpdate /InstallDir:C:\ruby24\msys64'" \
+&& refreshenv \
+&& ridk install 2 3 \
+&& echo gem: --no-document >> C:\ProgramData\gemrc \
+&& gem install --file 'C:\fluentd\install\Gemfile' \
+&& gem sources --clear-all
+
+RUN powershell -Command $ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue'; \
```
Comment (orangedeng) [inline package/windows/Dockerfile]: Is there any chance we can merge all the `RUN`? So that we can reduce the layer of the fluentd image.

Comment (aiwantaozi) [reply package/windows/Dockerfile]: The dockerfile is reference from fluentd's dockerfile, in order to sync with fluentd, I prefer we do like they do. https://github.com/fluent/fluentd-docker-image/blob/master/v1.6/windows/Dockerfile

**Comment by orangedeng** [comment]
LGTM

File: package/windows/Gemfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+gem 'json'
+
+# windows
+gem 'win32-service'
+gem 'win32-ipc'
+gem 'win32-event'
+gem 'windows-pr'
+
+# Input / Output plugins:
+gem 'fluent-plugin-elasticsearch'
```
Comment (gitlawr) [inline package/windows/Gemfile]: Why not specify the version requirement for plugins like we did for Linux?

File: package/windows/Gemfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+gem 'fluent-plugin-splunk-enterprise'
+gem 'fluent-plugin-kafka'
+gem 'fluent-plugin-kubernetes_metadata_filter'
+gem 'fluent-plugin-remote_syslog_tls'
+
+# filter plugins
+gem 'fluent-plugin-rewrite-tag-filter'
+
+# third party
+# gem 'zookeeper'
```
Comment (gitlawr) [inline package/windows/Gemfile]: Is this useless?

**Comment by aiwantaozi** [comment]
Linux is going to upgrade versions, the version will specify in gemfile.lock. Here are the related linux PR https://github.com/rancher/fluentd/pull/14

**Comment by gitlawr** [comment]
LGTM

**Comment by loganhz** [comment]
We need to remove all `-head` stuff

**Comment by loganhz** [comment]
LGTM

**Pull request merged**